src/Core/Addon/Module/ModuleManager.php

```diff
@@ -104,12 +104,8 @@ public function install($name)
         }
 
         if ($this->moduleProvider->isInstalled($name)) {
-            throw new Exception(
-                $this->translator->trans(
-                    'The module %module% is already installed.',
-                    array('%module%' => $name),
-                    'Admin.Modules.Notification'
-                ));
+            $module = $this->moduleRepository->getModule($name);
+            $module->onUninstall();
```

Comment:
No no no ! Do not uninstall the module during an upgrade ! It may delete some data.